libs/labelbox/src/labelbox/schema/disconnected_task.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, List, Optional, Protocol, Union
+
+from labelbox.exceptions import OperationNotSupportedException, ResourceNotFoundError
+from labelbox.orm.db_object import experimental
+from labelbox.orm.model import Field
+from labelbox.schema.task import Task, DataUpsertTask
+
+TaskDataType = Optional[Union[Dict[str, Any], List[Dict[str, Any]]]]
+
+
+class SupportsTaskQueries(Protocol):
+
+    @property
+    def status(self) -> Field:
+        ...
+
+    @property
+    def errors(self) -> TaskDataType:
+        ...
+
+    @property
+    def result(self) -> TaskDataType:
+        ...
+
+
+class TaskFactory:
+
+    @classmethod
+    def get(cls, client, task_id) -> SupportsTaskQueries:
```

Comment:
This method is redundant as it has the same method parameters as `get_task_by_id`.

I would not include both as it will only create confusion.